packages/contracts/src/deploy-utils.ts

```diff
@@ -53,9 +53,18 @@ export const deployAndVerifyAndThen = async ({
           address: result.address,
           constructorArguments: args,
         })
-        console.log('Successfully verified')
+        console.log('Successfully verified on Etherscan')
       } catch (error) {
-        console.log('Error when verifying bytecode on etherscan:')
+        console.log('Error when verifying bytecode on Etherscan:')
+        console.log(error)
+      }
+
+      try {
+        console.log('Verifying on Sourcify...')
+        await hre.run('sourcify')
```

Comment:
This required address to verify and constructor arguments, same as etherscan.

```
await hre.run("sourcify", {
                address: contract,
                constructorArguments: [...]
            });
```

Reply:
Interesting. I might be missing something, but it looks like the [`sourcify` task grabs the address and relevant metadata from `hre.deployments`](https://github.com/wighawag/hardhat-deploy/blob/2b1ade5b69b740b0d808a04370b31fa2e61d306f/src/sourcify.ts#L37-L115).